docs/nextflow/bundle-scripts.md

```diff
@@ -0,0 +1,84 @@
+---
+title: Bundling pipeline scripts
+description: Learn how Wave bundles scripts from Nextflow pipelines into container images
+date created: 2025-11-19
+date edited: 2025-11-19
+tags: [nextflow, wave, scripts, guides]
+---
+
+On cloud platforms without shared file systems, containerized tasks cannot access scripts from a shared directory. Wave solves this by packaging certain scripts directly into the container image and making them available in the task execution environment.
+
+This guide explains which `bin` script directories Wave bundles when building container images.
+
+## Script location and behavior
+
+Nextflow pipelines can store `bin` scripts in multiple locations. Wave treats each location differently:
+
+| Location                                                   | Nextflow project path             | Wave behavior         |
+|------------------------------------------------------------|-----------------------------------|-----------------------|
+| [Project `bin` directory](#project-bin-directory)          | `${projectDir}/bin/`              | Bundled conditionally |
+| [Module `bin` directory](#module-bin-directory)            | `${moduleDir}/resources/usr/bin/` | Always bundled        |
+
+### Project `bin` directory
+
+Wave does not bundle scripts from the project [`bin` directory](https://nextflow.io/docs/latest/sharing.html#the-bin-directory) by default. Nextflow uploads these scripts to the work directory at runtime using cloud APIs. This approach adds network overhead and can be inefficient when launching many tasks.
+
+When you enable Fusion or use the AWS Fargate executor, Wave bundles scripts from the project `bin` directory into a container layer. This provides better performance but ties scripts to specific execution environments.
```

Comment:
Why does Wave bundle for these two scenarios? Re: Fargate, does it matter if Fargate is being used for the head job only? Compute jobs only? Head and compute jobs?